templates/nextflow-s3-template.yaml

```diff
@@ -0,0 +1,162 @@
+              - "s3:ListBucket*"
+              - "s3:PutBucket*"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - Sid: AdminFullAccess
+            # gives admin full access to bucket
+            Effect: Allow
+            Principal:
+              AWS:
+                - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+                - Fn::ImportValue: !Sub "${AWS::Region}-workflows-nextflow-ci-service-account-ServiceRoleArn"
+                - !FindInMap [Accounts, !Ref AWS::AccountId, AdminRole]
+            Action:
+              - "s3:*"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - Sid: ForgeUserAccess
+            # gives forge user access to bucket
+            Effect: Allow
+            Principal:
+              AWS:
+                - Fn::ImportValue: !Sub "${AWS::Region}-nextflow-forge-service-user-ServiceUserArn"
+            Action:
+              - "s3:ListBucket"
+              - "s3:GetObject"
+              - "s3:PutObject"
+              - "s3:PutObjectTagging"
+              - "s3:DeleteObject"
+            Resource:
+              - !Sub arn:aws:s3:::${Bucket}
+              - !Sub arn:aws:s3:::${Bucket}/*
+
+          - !If
+            - HasReadWriteAccessArns
+            - Sid: ReadWriteAccess
+              # gives grantees read/write write access to the bucket (objects) with required --acl
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadWriteAccessArns
+              Action:
+                - "s3:*Object*"  
+              Condition:
+                StringEquals:
+                  s3:x-amz-acl: bucket-owner-full-control
+              Resource:
+                - !Sub arn:aws:s3:::${Bucket}/*   
+            - !Ref AWS::NoValue 
+
+          - !If
+            - HasReadWriteAccessArns
+            - Sid: ReadWriteBucketAccess
+              # gives grantees with read/write access read access to the bucket
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadWriteAccessArns
+              Action:
+                - "s3:AbortMultipartUpload"
+                - "s3:ListMultipartUploadParts"
+                - "s3:ListBucket*"
+                - "s3:GetBucketLocation"   
+              Resource:
+                - !Sub arn:aws:s3:::${Bucket}
+                - !Sub arn:aws:s3:::${Bucket}/*
+            - !Ref AWS::NoValue 
+
+          - !If
+            - HasReadOnlyAccessArns
+            - Sid: ReadOnlyAccess
+              # gives grantees read (only) access to the bucket
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadOnlyAccessArns
+              Action:
+                - "s3:ListBucket*"
+                - "s3:GetBucketLocation"
+                - "s3:GetObject"
+                - "s3:GetObjectAcl"
+                - "s3:AbortMultipartUpload"
```

Comment:
Why give AbortMultipartUpload to read-only user?

Reply:
ListMultipartUpload and AbortMultipartUpload were given to the read-only user [here](https://github.com/Sage-Bionetworks/aws-infra/blob/8e98fb5727027df0b22bf688f21b49c254391758/templates/s3-bucket-v2.j2#L196) -- but can be removed if they aren't needed in this case!

Reply:
Perhaps @zaro0508 can shed light on why that permission is given to read-only

Reply:
@daisyhan97 The policy you referenced was crafted by @brucehoff specifically for users who want direct access  to [synapse custom storage locations](https://help.synapse.org/docs/Custom-Storage-Locations.2048327803.html).  I'm not sure if you are trying to accomplish the same use case here but all info and test for that use case are in issue IT-1295

Reply:
Thank you for clarifying, @zaro0508!
@tthyer I wasn't able to find the reasoning behind the MultipartUpload permissions in IT-1295, so I removed both the MultipartUpload permissions from the ReadOnly users!